### DISCUSSION TITLE
The CoderDojo logo :)

### DISCUSSION THREAD

**Discussion by paulolc** [zalo/CascadeStudio#60] [Show and tell]
I had a blast creating the CoderDojo logo. For sure I'll use this in one of our sessions. It's a lot of fun! I'm sure the kids will simply love it.

(The code needs some work though! ^_^)

Anyway, here it is:

![image](https://user-images.githubusercontent.com/673004/105425200-cc4a0f00-5c40-11eb-8780-81650c44bbc7.png)

```javascript
//
// Here's the CoderDojo logo
//
// The community of 2333 free, open and local programming clubs for young people
// 58,000 young people are being creative with technology with the help of 12,000 volunteers 
// in 114 countries. Join us!
//
// https://coderdojo.com/
//

let cylmain = Cylinder(50,10)
let cyloutside = Cylinder(51,10)
let cylover = Cylinder(52,10)

let c1 = Cylinder(25,10)
let t1 = Translate([-25, 0, 0], c1 )
let c2 = Cylinder(25,10)
let t2 = Translate([25, 0, 0], c2 )

let halfcylinside = Difference(cylmain,  [ Translate([-50, -50, 0] , Box(100,50,10) )] )



let oneX = -27
let oneY = 0 
let oneScale = 1.4


let one = Union( [
    Translate([(oneX)*oneScale, (oneY-1)*oneScale, 0], Box(15*oneScale,2*oneScale,10) )
    ,Translate([(oneX)*oneScale, (oneY-5)*oneScale, 0], Box(2*oneScale,10*oneScale,10) )
    ,Translate([(oneX+15)*oneScale, (oneY-1)*oneScale, 0], Box(2*oneScale,6*oneScale,10) )
    ,Translate([(oneX+1)*oneScale, (oneY-6)*oneScale, 0], Box(6*oneScale,2*oneScale,10) )
    , Translate([(oneX+7)*oneScale, (oneY-5)*oneScale, 0], Cylinder(1*oneScale,10) )
    ,Translate([(oneX+16)*oneScale, (oneY+5)*oneScale, 0], Cylinder(1*oneScale,10) )
    ,Translate([(oneX+1)*oneScale, (oneY+5)*oneScale, 0], Cylinder(1*oneScale,10) )
    ,Translate([(oneX+1)*oneScale, (oneY-5)*oneScale, 0], Cylinder(1*oneScale,10) )
])

let zeroX = 10
let zeroY = -5
let zeroScale = 1.4

let height = 16
let width = 9

let zero = Union([ 
    Translate([(zeroX+1)*zeroScale, zeroY*zeroScale, 0], Box((height-1)*zeroScale,2*zeroScale,10) )
    ,Translate([(zeroX+1)*zeroScale, (zeroY+width)*zeroScale, 0], Box((height-1)*zeroScale,2*zeroScale,10) )
    ,Translate([(zeroX)*zeroScale, (zeroY+width+1)*zeroScale, 0], Box(2*zeroScale,(-width)*zeroScale,10) )
    ,Translate([(zeroX+height-1)*zeroScale, (zeroY+width+1)*zeroScale, 0], Box(2*zeroScale,(-width)*zeroScale,10) )
    ,Translate([(zeroX+1)*zeroScale, (zeroY+1)*zeroScale, 0], Cylinder(1*zeroScale,10) )
    ,Translate([(zeroX+height)*zeroScale, (zeroY+1)*zeroScale, 0], Cylinder(1*zeroScale,10) )
    ,Translate([(zeroX+1)*zeroScale, (zeroY+width+1)*zeroScale, 0], Cylinder(1*zeroScale,10) )
    ,Translate([(zeroX+height)*zeroScale, (zeroY+width+1)*zeroScale, 0], Cylinder(1*zeroScale,10) )
])

let yinyang = Difference( cylover , [ Difference( cyloutside , [ halfcylinside, t2 ]) , t1, zero ])

```

**Comment by Irev-Dev** [comment]
Nice @paulolc,

Do you know if the `1` and `0` in the original logo are from a particular font?

Because under the hood CascadeStudio uses [opentype.js](https://github.com/opentypejs/opentype.js#readme) to generate paths for fonts in order to extrude them for the `Text3D` function. It might be possible to expose this a little more so that custom fonts can be used so that you don't need to code the text so manually.

**Comment by paulolc** [comment]
Thanks for your comment, @Irev-Dev! 

Yes, indeed. According to CoderDojo Style Guide, the logo font for the 0 and 1 is OCR-A
https://company-51033.frontify.com/d/E6KNDhunr9mR/coderdojo-style-guide-1460385526 

Maybe I'll explore the use of the Text3D, then. :)

Anyway, creating the logo numbers ended up being an example of a good, simple and fun exercise to use the combination of just a couple of simple shapes and the Union operation. It also allows to introduce the concepts of scale and parameterized dimensioning of the shapes. The API is simple and easily understandable. CascadeStudio is fun! :)

**Reply by Irev-Dev** [reply]
Yeah cool @paulolc. Really good you're enjoying it. 
I made a sneaky [branch](https://github.com/Irev-Dev/CascadeStudio/pull/10) that added OCR-A font 🙃.

[Here's a demo](https://cascade-studio-git-kurt-ocr-a.irev-dev.vercel.app/?code=C0mtKDF20VAyUNBVMFTSUTA20VEw0DPUUVD3dw7SdVTX5AIA&gui=q1ZKzs8tyM9LzSvxS8xNVbJSSk4sTk5MSQ3LTC1X0lHyTS3OCEotVrIy0DOE84IS89KBSqMN9AwMdYxidZScE5MzUu2VrEqKSlN1lIISUzJLgVqMDWBsmAYjAx0Tg9haAA%3D%3D)

CascadeStudio already allows for loading stl etc files, might be worth adding font upload too.

**Reply by paulolc** [reply]
Nice! Ok, so here it is then. [The CoderDojo logo using Text3D with OCR-A](https://cascade-studio-git-kurt-ocr-a.irev-dev.vercel.app/?code=jVLBjpswEL3nK6a5FCQnGNio1VY9tMmh6qXSKreIg5cMwStjI2PSpV%2FfMSFp2CzbtSwkz3vPnjePKJpFEfxAix8bcCXC2uzRbsyTAWUOhkCPbwnITVW1WroOTAFJmqZQWEQGpkYNQu%2BJnwsFtTUHK6pK6gPkqn1soDAWOtPSuUZTK%2FQXrj4zzvmoDMIiPGKvsyicPCL8lq4Eh3mpDXXTDWdqpkRV%2Bz7ipL%2FnaFSrHaJtwN8uNcTxHXVMRSuxWcJPQ7W2%2BTD4KZ2rm%2Fsoyr3ZPZldkjuPzGYKHeSdqgQJvsK6U1ITJ1hxFvPwjJrWNXKPI0I8IhzRjtCkR09wfI0kq4vOeWBrhW6UcBjsFoQBp50xLxouT6bUyVh9LU5geLsUqsi9eGh%2FI4uCstc5BmfXDGAGE2s3Dfn173XYLWhi0H949rbKLwbfzXMQc85Og35bEU6iE08N7jupO0H%2F143vPi5G9l7Wh5w9NBod8%2BPOQgIotGy4%2Fg9aM84g9vHQzm66Yg%2FG9RTOiJCx5BN%2FtfMLLyYen%2Bb13C0%2Bu3QTzPmcpcRe3rH5r%2FXD4tv89YHdVseV8MvJl9E4skXhpmdj43m%2Fx9V7HTE424n%2Fb%2BdF45dTCOThOtThF%2FCB9nFRdn8B&gui=q1ZKzs8tyM9LzSvxS8xNVbJSSk4sTk5MSQ3LTC1X0lHyTS3OCEotVrIy0DOE84IS89KBSqMN9AwMdYxidZScE5MzUu2VrEqKSlN1lIISUzJLgVqMDWBsmAYjAx0Tg9haAA%3D%3D)

Yeah, font upload would be cool.

```javascript
//
// Here's the CoderDojo logo using Text3D and the OCR-A preloaded font
//
// The community of 2333 free, open and local programming clubs for young people
// 58,000 young people are being creative with technology with the help of 12,000 volunteers 
// in 114 countries. Join us!
//
// https://coderdojo.com/
//

let cylmain = Cylinder(50,10)
let cyloutside = Cylinder(51,10)
let cylover = Cylinder(52,10)

let c1 = Cylinder(25,10)
let t1 = Translate([-25, 0, 0], c1 )
let c2 = Cylinder(25,10)
let t2 = Translate([25, 0, 0], c2 )

let halfcylinside = Difference( cylmain,  
                        [ 
                            Translate( [-50, -50, 0] 
                                , Box(100,50,10) 
                            )
                        ] 
                    )

let yinyang = Difference( cylover , [ Difference( cyloutside , [ halfcylinside, t2 ]) , t1 ])

let zero = Translate([15,10,10]
                ,Rotate([0,0,1],270
                    ,Rotate([1,0,0],270
                        ,Text3D("0",30,0.4,"OCR-A")
                    )
                )
            );

let one = Translate( [-35,10,10] 
            ,Rotate([0,0,1],270
                ,Rotate([1,0,0],270
                    , Text3D("1",30,0.4,"OCR-A")
                )
            )
        ) ;

Difference( yinyang, [ zero ])

```

**Reply by Irev-Dev** [reply]
Nice, much simpler.

I modified your original a little to print it with a base.
![IMG_4484](https://user-images.githubusercontent.com/29681384/105616866-e8f76a00-5e2d-11eb-93ea-5906b50d2814.JPG)

It's so thin and flexible I think with double-sided tape it could be used as a sticker.

Are you on reddit @paulolc? I was thinking of sharing it on the [coderdojo](https://www.reddit.com/r/coderdojo/) subreddit, but I'll let you if you'd prefer (feel free to take the above image).

**Reply by paulolc** [reply]
No, please go ahead! That's an excellent idea. I will sure upvote it ;) . I use reddit but I didn't know the CoderDojo community was so active there. Go figure :) I usually hang out in their Slack channels, but I'm definitely check the reddit now.

That 3D print looks so nice. We could use something like that in a swag giveway when we restart our sessions (when our lockdown is over, of course)

**Reply by Irev-Dev** [reply]
I does not look super active tbh, but I posted there anyway. https://www.reddit.com/r/coderdojo/comments/l3oj9q/paulolcs_coder_dojo_logo_model_in_cascadestudio/
Also tagged you on slack :)

Yeah that's a good idea @paulolc, if you make it as thin as I did it prints quickly too, they could probably afford to be a little smaller, bit you'd be able to fill an entire print bed with them and get 20+ of them in one go.

I was just thinking to myself I wonder how they're doing with lockdowns. I can see there's a channel on the slack "online_sessions" so I guess that's how you're dealing with it. Not the same but good given the circumstances.

**Reply by Irev-Dev** [reply]
Post again if/when you run a session. Would love to hear how it goes.
